api/internalutils/stream/stream_test.go

```diff
@@ -506,6 +506,138 @@ func TestTake(t *testing.T) {
 	}
 }
 
+// TestSkip tests the Skip combinator.
+func TestSkip(t *testing.T) {
+	t.Parallel()
+
+	// normal usage
+	s, err := Collect(Skip(Slice([]int{1, 2, 3, 4}), 2))
+	require.NoError(t, err)
+	require.Equal(t, []int{3, 4}, s)
+
+	// skip all
+	s, err = Collect(Skip(Slice([]int{1, 2, 3, 4}), 4))
+	require.NoError(t, err)
+	require.Empty(t, s)
+
+	// skip none
+	s, err = Collect(Skip(Slice([]int{1, 2, 3, 4}), 0))
+	require.NoError(t, err)
+	require.Equal(t, []int{1, 2, 3, 4}, s)
+
+	// negative skip
+	s, err = Collect(Skip(Slice([]int{1, 2, 3, 4}), -1))
+	require.NoError(t, err)
+	require.Equal(t, []int{1, 2, 3, 4}, s)
+
+	// skip more than available
+	s, err = Collect(Skip(Slice([]int{1, 2, 3, 4}), 5))
+	require.NoError(t, err)
+	require.Empty(t, s)
+
+	// positive skip on empty stream
+	s, err = Collect(Skip(Empty[int](), 2))
+	require.NoError(t, err)
+	require.Empty(t, s)
+
+	// zero skip on empty stream
+	s, err = Collect(Skip(Empty[int](), 0))
+	require.NoError(t, err)
+	require.Empty(t, s)
+
+	// negative skip on empty stream
+	s, err = Collect(Skip(Empty[int](), -1))
+	require.NoError(t, err)
+	require.Empty(t, s)
+
+	// immediate failure
+	err = Drain(Skip(Fail[int](fmt.Errorf("unexpected error")), 1))
+	require.Error(t, err)
+
+	// failure during skip
+	err = Drain(Skip(Chain(
+		Slice([]int{1, 2}),
+		Fail[int](fmt.Errorf("unexpected error")),
+		Slice([]int{3, 4}),
+	), 3))
+	require.Error(t, err)
+}
+
+// TestFlatten tests the Flatten combinator.
+func TestFlatten(t *testing.T) {
+	t.Parallel()
+
+	// normal usage
+	s, err := Collect(Flatten(Slice([]Stream[int]{
+		Slice([]int{1, 2}),
+		Slice([]int{3, 4}),
+		Slice([]int{5, 6}),
+	})))
+	require.NoError(t, err)
+	require.Equal(t, []int{1, 2, 3, 4, 5, 6}, s)
+
+	// empty stream
+	s, err = Collect(Flatten(Empty[Stream[int]]()))
+	require.NoError(t, err)
+	require.Empty(t, s)
+
+	// empty substreams
+	s, err = Collect(Flatten(Slice([]Stream[int]{
+		Empty[int](),
+		Slice([]int{1, 2, 3}),
+		Empty[int](),
+		Slice([]int{4, 5, 6}),
+		Empty[int](),
+	})))
+	require.NoError(t, err)
+	require.Equal(t, []int{1, 2, 3, 4, 5, 6}, s)
+
+	// immediate failure
+	err = Drain(Flatten(Fail[Stream[int]](fmt.Errorf("unexpected error"))))
+	require.Error(t, err)
+
+	// failure during streaming
+	err = Drain(Flatten(Slice([]Stream[int]{
+		Slice([]int{1, 2}),
+		Fail[int](fmt.Errorf("unexpected error")),
+		Slice([]int{3, 4}),
+	})))
+	require.Error(t, err)
```

Comment:
We can verify the results are as expected:

```suggestion
	// failure during streaming
	s, err = Collect(Flatten(Slice([]Stream[int]{
		Slice([]int{1, 2}),
		Fail[int](fmt.Errorf("unexpected error")),
		Slice([]int{3, 4}),
	})))
	require.Error(t, err)
	require.Equal(t, []int{1, 2}, s)
```